paasta_tools/frameworks/native_scheduler.py

```diff
@@ -422,7 +406,7 @@ def kill_tasks_if_necessary(self, driver):
         )
         new_tasks_to_kill = new_tasks_by_desirability[desired_instances:]
 
-        old_tasks = self.get_old_tasks(base_task.name, self.tasks_with_flags.keys())
+        old_tasks = self.get_old_tasks(base_task.name, self.task_store.get_all_tasks().keys())
```

Comment:
now that this could be somewhere behind the network it's probably worth caching the first call to `get_all_tasks` and reusing here and on line 427?